src/ImmutableVirtualizedList/ImmutableVirtualizedList.js

```diff
@@ -26,14 +25,31 @@ class ImmutableVirtualizedList extends PureComponent {
     immutableData: (props, propName, componentName) => {
       // Note: It's not enough to simply validate PropTypes.instanceOf(Immutable.Iterable),
       // because different imports of Immutable.js across files have different class prototypes.
-      // TODO: Add support for Immutable.Map, etc.
-      if (Immutable.Map.isMap(props[propName])) {
-        return new Error(`Invalid prop ${propName} supplied to ${componentName}: Support for Immutable.Map is coming soon. For now, try an Immutable List, Set, or Range.`);
-      } else if (!utils.isImmutableIterable(props[propName])) {
+      if (!utils.isImmutableIterable(props[propName])) {
         return new Error(`Invalid prop ${propName} supplied to ${componentName}: Must be instance of Immutable.Iterable.`);
       }
     },
 
+    /**
+     * Whether to treat immutableData as a sectioned list,
+     * applying section headers.
+     */
+    treatAsSections: PropTypes.bool,
+
+    /**
+     * A function that returns some {@link PropTypes.element}
+     * to be rendered as a section header. It will be passed a List
+     * or Map of the section's items, and the section key.
+     */
+    renderSectionHeader: PropTypes.func,
+
+    /**
+     * A function that returns some {@link PropTypes.element}
+     * to be rendered as a section row. It will be passed the item,
+     * and the item's key.
+     */
+    renderRow: PropTypes.func,
```

Comment:
Can we just leave it as `renderItem` (i.e. remove this prop)?

Reply:
Yep I think so.